packages/core/src/types.ts

```diff
@@ -1325,3 +1328,23 @@ export type InterpreterFrom<
 >
   ? Interpreter<TContext, TStateSchema, TEvent, TTypestate>
   : never;
+
+export interface ActorContext<TEvent extends EventObject, TEmitted> {
+  parent?: ActorRef<any, any>;
+  self: ActorRef<TEvent, TEmitted>;
+}
+
+export type LifecycleSignal =
+  | {
+      type: 'xstate.start';
+    }
+  | { type: 'xstate.stop' };
+
+export interface Behavior<TEvent extends EventObject, TEmitted = any> {
+  transition: (
+    state: TEmitted,
+    event: TEvent | LifecycleSignal,
+    actorCtx: ActorContext<TEvent, TEmitted>
+  ) => TEmitted;
```

Comment:
Q: do u still plan to introduce transducers anyhow? if yes - how they will relate to behaviors? should behavior's `transition` be pure?

Reply:
Will have to think about this. The simplest approach is that the actions _are_ in the state (just like in a machine), but then we need to tell whatever is interpreting the behavior that those actions should be executed 🤔